#1099 remove create listing applicant

ephraimdjeket opened the pull request:
## Title
[#1099 remove create listing applicant](https://github.com/NoroffFEU/agency.noroff.dev/tree/%231099-remove-create-listing-applicant)

## Describe your changes: 
Implemented a new function to hide the anchor tag link "new listing +" which would redirect users to a new page where they could create a new listing. This feature was available for both roles and after reviewing it, the conclusion was that it was unnecessary for users with the role "Applicant" to have it visible as it would lead to an error with the status code: 401 unauthorized.

## Type of changes:

- Added a new function to check whether user logged in is of the role "Client" or "Applicant" to hide/show an anchor tag.
- Created a test file for the new logic to ensure that everything is working as expected.

## Checklist before requesting a review

* [x] I created a new branch before starting with this ticket?
* [x] I commented the issue after finish in the Frontend Team Board?
* [x] I moved the card to Quality Assurance after finish?

## Did you run into any issues?
Upon fetching some data in the localStorage, I noticed that there were some invisible characters added to the value side, which gave me some trouble. Spent some time troubleshooting.

## Screenshots:
- Yes, the screenshots are included in the table below.

| Before | After |
---------|--------
![image](https://github.com/user-attachments/assets/651ce802-4751-40dd-99cd-1c5967d82140) | ![image](https://github.com/user-attachments/assets/d7a08372-6962-49e2-a56a-f0cafe4aa55f)

## Feedback
- Yes, I want feedback.

ddyrnes commented on src/js/ui/user/createListingVisibility.js:
```diff
@@ -0,0 +1,23 @@
+  const isCompany = localStorage.getItem('role');
+
+  if (newListing) {
+    if (isCompany === '"Client"') {
```
Both '"Client"' and '"Applicant"' is using both single and double quotes, is there a reason for this?

ephraimdjeket replied:
The reason for this is because both "Client" and "Applicant" are not parsed, which is why they're compared to in double quotes as string. 

I've refactored it by parsing the key value of role into a javascript value and removed the double quotes for a cleaner look and consistency with other files.

ddyrnes reviewed (commented):
Hi @ephraimdjeket 
I reviewed your code and ran the tests. 
The tests all passed and are well setup. Good job!
When I checked your branch vs the main, it looks like you fixed the issue with applicants seeing "New listing +", which is now hidden/gone on the PR branch.
But it seems like it has created some other issues.

If you see the provided screenshot, this is comparing main to your branch. 

It looks like favourite listings are removed from the main view, and placed under "application". Is this intended?
There also seems to be an issue with background of favourite listings being dark making the title unreadable, while on hover it turns regular again.

Can you test this on your end to see if you get same result and get back to me? 
![pr1127](https://github.com/user-attachments/assets/2a97d4c0-dab9-448c-9682-b745dcd4bddf)

ephraimdjeket commented:
Hi @ddyrnes!

Thank you for good review and valuable feedback! After looking over the files again it seems like I might've deleted more than I should've. I've now fixed the issue, I've refactored some code that was mentioned above, ran all tests before pushing and tested the updated files via vite.

Could you review again and give me feedback? Thank you.

| Offers | Applications |
---------|--------
![image](https://github.com/user-attachments/assets/fc4d5de8-0cf0-496d-8459-d7d812b04193) | ![image](https://github.com/user-attachments/assets/85ac4f55-e247-4f9f-82f7-7e50a66e7847)
